pkg/kubectl/cmd/set/set_image.go

```diff
@@ -131,13 +131,13 @@ func (o *ImageOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []st
 		ContinueOnError().
 		NamespaceParam(cmdNamespace).DefaultNamespace().
 		FilenameParam(enforceNamespace, &o.FilenameOptions).
+		SelectorParam(o.Selector).
```

Comment:
From the godoc for the resource builder, it looks like it does not affect objects loaded from a file or a URL https://github.com/openshift/origin/blob/master/vendor/k8s.io/kubernetes/pkg/kubectl/resource/builder.go#L242